### PR TITLE
Add mark config properties to mark definition

### PR DESCRIPTION
**Pull request by kanitw**
Rationale: now that config is a top-level property -- there is no direct way to customize each individual single view.  

(The only possible way is to use `style`, but that is pretty indirect and make more sense when one want to apply the same styles to multiple parts.) 

An alternative solution would be implementing encoding channels for all of these properties. However, that requires careful thinking about scale and scale ranges + legends for all of the properties we may add.  Given that's a non-trivial amount of design work, I don't think it's gonna happen anytime soon so this seems like the most natural thing to do.

### CODE REVIEW

File: site/docs/mark/mark.md [hunk 1]
```diff
@@ -62,9 +62,9 @@ For example, a bar chart has `mark` as a simple string `"bar"`.
 }
 ```
 
-A mark definition object can contains the following values.
+A mark definition object can contains the following properties and all properties of the mark config:
```
Comment (domoritz) [inline site/docs/mark/mark.md]: Would it make sense to link to it or include all properties in the table?

Comment (kanitw) [reply site/docs/mark/mark.md]: Good catch.  I'll add a link

Comment (domoritz) [reply site/docs/mark/mark.md]: Ping me when you added it.

Comment (kanitw) [reply site/docs/mark/mark.md]: Done

**Pull request merged**